utils/mutex-utils.js

```diff
@@ -0,0 +1,69 @@
+const AWS = require('aws-sdk');
+const { serializeError } = require('serialize-error')
+
+const logger = require('../logger/logger')
+const { GetMutexError, ReleaseMutexError } = require('../errors/MutexError')
+
+// Env vars
+const { NODE_ENV, MUTEX_TABLE_NAME } = process.env
+const IS_LOCAL_DEV = NODE_ENV === 'LOCAL_DEV'
+const mockMutexObj = {}
+
+// Dynamodb constants
+const AWS_REGION_NAME = 'ap-southeast-1'
+AWS.config.update({region: AWS_REGION_NAME})
+const docClient = new AWS.DynamoDB.DocumentClient();
+
+const mockLock = (siteName) => {
```

Comment:
Using a local object is to store data is slightly different to how DynamoDB works - would we be open to using a local version of DynamoDB? https://medium.com/@vschroeder/install-a-local-dynamodb-development-database-on-your-machine-82dc38d59503

One issue I can see with this is that it's not straightforward to run tests for our CI/CD workflow since there aren't many GitHub actions for setting up local DynamoDB / there are issues associated with testing with the actual DynamoDB

Reply:
I meant to set it up as a hack for now (where CI would run a test against the staging DB) - but yes I'll do this instead

Reply:
Ah sorry I wasn't clear, I meant that we could do that in a subsequent PR

Reply:
### Update
We had a discussion about this and decided that the current set up is okay:
- local dev: the mutex table is represented by a local object
- test (CI test): use the staging DynamoDB mutex table (since the records should be transient)
- staging: use the staging DynamoDB mutex table
- prod: use the prod DynamoDB mutex table